### DISCUSSION TITLE
Plugin/Add-on system to broaden StreetComplete's use cases

### DISCUSSION THREAD

**Discussion by hb0nd** [streetcomplete/StreetComplete#3959] [Ideas]
**Use case**
an add-on system would be really cool, currently there are many different apps for contributing to different projects, for example the [Wikimedia Commons app](https://commons-app.github.io/). A plugin system would allow these quests to be displayed in StreetComplete. Another example of open source projects that could develop an add-on is [OpenBenches.](https://openbenches.org/)

This would bring much wider StreetComplete adoption, allow custom OSM-related functions, and also centralise mobile open-source project contribution 😄

**Comment by hb0nd** [comment]
Perhaps this could be a GSoC challenge

**Comment by 10992-osm** [comment]
Is there any particular advantage of add-ons over just adding new quests to the app?

**Reply by hb0nd** [reply]
> Is there any particular advantage of add-ons over just adding new quests to the app?

It'd mean that StreetComplete maintainers wouldn't have to maintain a bunch of additional stuff, instead that work would be offloaded to eh plugin maintainers. It'd also mean that anyone can make a plugin without having to convince the maintainers that it'd be useful to merge.

**Reply by 10992-osm** [reply]
> > Is there any particular advantage of add-ons over just adding new quests to the app?
> 
> It'd mean that StreetComplete maintainers wouldn't have to maintain a bunch of additional stuff, instead that work would be offloaded to eh plugin maintainers. It'd also mean that anyone can make a plugin without having to convince the maintainers that it'd be useful to merge.

One thing I have thought of is that the maintainers might wish to retain a certain amount of control in order to avoid damaging SC's reputation among the OSM community.

**Reply by severinmenard** [reply]
An advantage would be to avoid OSM experienced people proposing new quests not being added because considered as not interesting/not essential
Regarding "retain a certain amount of control in order to avoid damaging SC's reputation": JOSM both hosts external preset extensions in its server + allows anyone to add its own extension from a local file. And I do not think JOSM has any bad reputation.

**Comment by westnordost** [comment]
Do you know MapComplete? It sounds like this is what you are looking for. No reason to copy 1:1 the concept of another editor.

**Reply by mnalis** [reply]
link: https://wiki.openstreetmap.org/wiki/MapComplete

**Comment by matkoniecz** [comment]
If someone really wants they may implement something that scan for nearby SC quests and nearby places without photos etc.

But I expect that for SC it is not very useful as you will get quests basically everywhere.
